### PR TITLE
update witness class in elasticsearch-6.x-plugin

### PR DESCRIPTION
**Pull request by xzyJavaX**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes [#9258](https://github.com/apache/skywalking/issues/9258)
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java [hunk 1]
```diff
@@ -260,6 +260,6 @@ public StaticMethodsInterceptPoint[] getStaticMethodsInterceptPoints() {
 
     @Override
     protected String[] witnessClasses() {
-        return new String[] {Constants.TASK_TRANSPORT_CHANNEL_WITNESS_CLASSES};
+        return new String[] {Constants.CREATE_INDEX_RESPONSE_WITNESS_CLASSES};
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java:263]: So, **TASK_TRANSPORT_CHANNEL_WITNESS_CLASSES** as witness class is working for other interceptors?

**Comment by wu-sheng** [comment]
I think you break the CI, please recheck.

File: apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java (hunk 1 above)
Comment (xzyJavaX) [reply apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java:263]: There are no other problems for the time being.

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java:263]: What do you mean? Why does this matter and others are not?

I can see `org.elasticsearch.transport.TaskTransportChannel` is used for 6 instrumentation definitions, if this witness class is changed, but others are not, I would assume, there are still other interceptors are going to run?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java:263]: @xzyJavaX We need feedback about this before we merge this change.

Comment (xzyJavaX) [reply apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/define/RestHighLevelClientInstrumentation.java:263]: We confirm that TASK_TRANSPORT_CHANNEL_WITNESS_CLASSES is working for other interceptors.And the version of transport-client can be changed from 6.7.1-6.8.4 to 6.2.0-6.8.4.Test cases will be added later.

File: test/plugin/scenarios/elasticsearch-rest-high-level-6.x-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -307,124 +307,6 @@ segmentItems:
       - {key: db.type, value: Elasticsearch}
       - {key: db.instance, value: not null}
       skipAnalysis: 'false'
-    - operationName: Elasticsearch/IndexRequest
```
Comment (wu-sheng) [inline test/plugin/scenarios/elasticsearch-rest-high-level-6.x-scenario/config/expectedData.yaml:310]: Why do these get removed?

Comment (xzyJavaX) [reply test/plugin/scenarios/elasticsearch-rest-high-level-6.x-scenario/config/expectedData.yaml:310]: Moved to elasticsearch-transport-6.x-scenario

Comment (xzyJavaX) [reply test/plugin/scenarios/elasticsearch-rest-high-level-6.x-scenario/config/expectedData.yaml:310]: The version of rest-high-level-client changes too much, cannot be cover with a test case, in order not to affect the test of transport, so there are two cases.

Comment (wu-sheng) [reply test/plugin/scenarios/elasticsearch-rest-high-level-6.x-scenario/config/expectedData.yaml:310]: It is fine you don't add more versions to support.
Please update doc according to the latest status.

Comment (xzyJavaX) [reply test/plugin/scenarios/elasticsearch-rest-high-level-6.x-scenario/config/expectedData.yaml:310]: Supported-list.md updated.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**